bundle/edu.gemini.itc/src/main/scala/edu/gemini/itc/service/ItcServiceImpl.scala

```diff
@@ -26,8 +29,24 @@ class ItcServiceImpl extends ItcService {
 
   def calculate(source: SourceDefinition, obs: ObservationDetails, cond: ObservingConditions, tele: TelescopeDetails, ins: InstrumentDetails): Result = try {
 
-    if (obs.getMethod.isImaging)  calculateImaging(source, obs, cond, tele, ins)
-    else                          calculateSpectroscopy(source, obs, cond, tele, ins)
+    // Get the SED data from an aux file. For now we can assume that the ITC service is running on the same
+    // machine as the database (localhost). In case this setup changes, we need to change this here, too.
+    def auxFileDistribution(id: String, name: String): SpectralDistribution = {
+      val programId     = SPProgramID.toProgramID(id)
+      val spectrumBytes = new AuxFileClient("localhost", 8443).fetchToMemory(programId, name)
+      val spectrum      = new String(spectrumBytes)
```

Comment:
The good news is that these particular files usually are just long lists of numbers. However I agree that encoding can break things. My hope is, that `String(bytes[])` tries to do something sensible by using the system's default encoding or whatever, but of course I might be wrong about that and also if someone throws a file at us with a weird encoding then I really don't know how we can handle that gracefully. I'll do some additional testing and see what happens if we throw weird encodings at this.